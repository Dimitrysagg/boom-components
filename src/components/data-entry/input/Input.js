import React from 'react'
import PropTypes from 'prop-types'

import { DefaultInput } from './styles'

/**
 * Basic text field to get user input.
 *
 * @param {*} { size, ...props }
 * @returns {Component}
 */
const Input = ({ size, ...props }) => <DefaultInput size={size} {...props} />

Input.propTypes = {
  /** callback when value changes */
  onChange: PropTypes.func,
  /** size of the input box */
  size: PropTypes.oneOf(['small', 'default', 'large']),
  /** input content value */
  value: PropTypes.string
}

Input.defaultProps = {
  size: 'default'
}

export default Input
