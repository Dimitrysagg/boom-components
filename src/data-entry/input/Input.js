import React from "react";
import PropTypes from "prop-types";
import { noop } from "../../helpers";

import { DefaultInput } from "./styles";

/**
 * Basic text field to get user input.
 *
 * @param {*} { size, ...props }
 * @returns {Component}
 */
const Input = ({ size, ...props }) => <DefaultInput size={size} {...props} />;

Input.propTypes = {
  /** label text or component displayed on the right side of the input */
  addonAfter: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
  /** label text or component displayed on the left side of the input */
  addonBefore: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
  /** whether input content can be removed with clear icon */
  clearable: PropTypes.bool,
  /** initial input content */
  defaultValue: PropTypes.string,
  /** disabled state of the input */
  disabled: PropTypes.bool,
  /** callback when value changes */
  onChange: PropTypes.func,
  /** callback when enter key is pressed */
  onPressEnter: PropTypes.func,
  /** prefix icon inside input */
  prefix: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
  /** suffix icon inside input */
  suffix: PropTypes.oneOfType([PropTypes.string, PropTypes.node]),
  /** HTML type of input */
  type: PropTypes.oneOf([
    "button",
    "checkbox",
    "color",
    "date",
    "datetime",
    "datetime-local",
    "email",
    "file",
    "hidden",
    "image",
    "month",
    "number",
    "password",
    "radio",
    "range",
    "reset",
    "search",
    "submit",
    "tel",
    "text",
    "time",
    "url",
    "week"
  ]),
  /** size of the input box */
  size: PropTypes.oneOf(["small", "default", "large"]),
  /** input content value */
  value: PropTypes.string
};

Input.defaultProps = {
  addonAfter: null,
  addonBefore: null,
  clearable: true,
  defaultValue: null,
  disabled: false,
  onChange: noop,
  onPressEnter: noop,
  size: "default",
  type: "text",
  prefix: null,
  suffix: null,
  value: null
};

export default Input;
