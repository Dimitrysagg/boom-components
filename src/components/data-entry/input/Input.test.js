import React from 'react'
import { mount } from 'enzyme'

import Input from './Input'

describe('Input', () => {
  test('Snapshot', () => {
    const wrapper = mount(<Input value="Test" readOnly />)
    const tree = wrapper.find(Input)

    expect(tree).toMatchSnapshot()
  })
})
