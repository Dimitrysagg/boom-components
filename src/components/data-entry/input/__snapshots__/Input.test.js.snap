// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Snapshot 1`] = `
<Input
  disabled={false}
  readOnly={true}
  size="default"
  type="text"
  value="Test"
>
  <styled.input
    disabled={false}
    readOnly={true}
    size="default"
    type="text"
    value="Test"
  >
    <StyledComponent
      disabled={false}
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": false,
            "lastClassName": "cNTepu",
            "rules": Array [
              "
  display: inline-block;
  position: relative;
  border: 1px solid ",
              "#e9e9e9",
              ";
  border-radius: 4px;
  width: 200px;
  margin: 0 8px 8px 0;
  line-height: 1.5;
  padding: ",
              [Function],
              ";
  font-size: ",
              [Function],
              ";
  height: ",
              [Function],
              ";
",
            ],
          },
          "displayName": "styled.input",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "input",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
      readOnly={true}
      size="default"
      type="text"
      value="Test"
    >
      <input
        className="sc-bdVaJa cNTepu"
        disabled={false}
        readOnly={true}
        size="default"
        type="text"
        value="Test"
      />
    </StyledComponent>
  </styled.input>
</Input>
`;
